processor/ratelimitprocessor/ratelimiter.go

```diff
@@ -51,6 +52,20 @@ type RateLimiter interface {
 // high cardinality: tenant ID would be a good choice. For rate
 // limiting by IP (e.g. to avoid DDoS), consider running OpenTelemetry
 // Collector behind a WAF/API Gateway/proxy.
+
+type metrics struct {
+	ratelimitRequests metric.Int64Counter
+}
+
+func getProjectIDFromMetadata(ctx context.Context, key string) string {
+	clientInfo := client.FromContext(ctx)
+	values := clientInfo.Metadata.Get(key)
+	if len(values) > 0 {
+		return values[0]
+	}
+	return "" // or a default value
+}
```

Comment:
```suggestion
func getProjectIDFromMetadata(ctx context.Context) (string, error) {
	clientInfo := client.FromContext(ctx)
	values := clientInfo.Metadata.Get(projectIDKey)
	if len(values) > 0 {
		return values[0], nil
	}
	return "", errors.New("project id not found") // or a default value
}
```

What do you think? Hardcode the key to be the elastic project id, and return error if not found

Reply:
Add it as well. I leave this comment open to review because now if we dont receive projectID then we return with error. I dont know if we want to be so strict. Maybe we can have the metrics but without enhancement?

Reply:
Small update on this, I only set projectID "" when key not found. This was breaking other tests we had where we dont set the specific metadata key. I think we are ok by setting to empty